### PR TITLE
Require libical >= 3.0.10

### PR DESCRIPTION
**Pull request by ksmurchison**

### CODE REVIEW

File: imap/ical_support.h [hunk 1]
```diff
@@ -203,14 +203,6 @@ extern void icalcomponent_normalize_x(icalcomponent *ical);
...
     icalcomponent_get_first_property(comp, ICAL_ACKNOWLEDGED_PROPERTY)
 
-#ifndef HAVE_RFC7986_COLOR
-
-/* Replacement for missing function in 3.0.0 <= libical < 3.0.5 */
-
-extern icalproperty *icalproperty_new_color(const char *v);
-
-#endif /* HAVE_RFC7986_COLOR */
-
```
Comment (elliefm) [inline imap/ical_support.h:213]: Should we also remove the implementation of this from [ical_support.c](https://github.com/cyrusimap/cyrus-imapd/blob/master/imap/ical_support.c#L2121-L2132), or does it need to remain for some reason I don't understand?

**Review by elliefm — COMMENTED** [review]
Thanks! One question...

**Comment by elliefm** [comment]
The CI is failing on these tests:

* JMAPCalendars.calendarevent_set_fullblown                                            
* JMAPCalendars.calendarevent_get_privacy_shared

and both are failing because they're missing an expected "color" property.  Hmm.

The build succeeded, so the docker image must have a new enough libical.  I wonder if something else is conditional on that `HAVE_RFC7986_COLOR` which is no longer being defined during configure?  But when I search for that string, the only occurrences are the two in ical_support.c that I already identified.  So I'm puzzled by why those two tests should suddenly not get their expected "color" fields anymore.

Another hmm: I can reproduce those two cass failures locally, so that'll make it easier to figure out what's up.  I don't expect a response from you till you're back from leave, so I'll tinker in the meantime, and if I don't figure it out myself, it'll wait.

**Comment by elliefm** [comment]
Oh curious: in both cases the failure looks something like this:

```
 Structures begin differing at:
  $a->{recurrenceOverrides}{2021-01-02T01:00:00}{color} = undef
  $b->{recurrenceOverrides}{2021-01-02T01:00:00}{color} = Does not exist
 at tiny-tests/JMAPCalendars/calendarevent-get-privacy-shared line 199.
```

That is, one of the deep structures being compared has the "color" key with an undefined value, and the other one doesn't have the "color" key at all.  It's not clear to me at a glance which of these is the "actual" vs the "expected".

**Comment by elliefm** [comment]
And it's specifically inside "reccurranceOverrides", not at the higher level.

**Comment by elliefm** [comment]
In JMAPCalendars.calendarevent_set_fullblown, we create an event that has this:

```
        recurrenceOverrides => {
            '2021-02-02T02:00:00' => {
                title => 'recurrenceOverrideTitle',
            },
        },
```

and get this back in the response:

```
  'recurrenceOverrides' => {
    '2021-02-02T02:00:00' => {
      'color' => undef,
      'title' => 'recurrenceOverrideTitle',
    },
  },
```

And then the test compares the original with the result, and complains because it didn't expect a `'color'` key to be present.

But I don't understand why this would have changed from this change...

**Comment by elliefm** [comment]
Oh wild: if I remove the old fallback `icalproperty_new_color()` implementation from ical_support.c and rebuild, like this:

```
diff --git a/imap/ical_support.c b/imap/ical_support.c
index 5003c2ade..104de592d 100644
--- a/imap/ical_support.c
+++ b/imap/ical_support.c
@@ -2117,20 +2117,6 @@ EXPORTED int icalcomponent_apply_vpatch(icalcomponent *ical,
     return 0;
 }
 
-
-#ifndef HAVE_RFC7986_COLOR
-
-/* Replacement for missing function in 3.0.0 <= libical < 3.0.5 */
-
-EXPORTED icalproperty *icalproperty_new_color(const char *v)
-{
-    icalproperty *prop = icalproperty_new_x(v);
-    icalproperty_set_x_name(prop, "COLOR");
-    return prop;
-}
-
-#endif /* HAVE_RFC7986_COLOR */
-
 EXPORTED const char *icaltimezone_get_location_tzid(const icaltimezone *zone)
 {
     const char *v = icaltimezone_get_location((icaltimezone*) zone);
```

then both tests start passing.  So now I understand:

* the fallback `icalproperty_new_color()` had a different behaviour from the libical one when its argument was NULL, but we never noticed because we always have new enough libical that we don't use the fallback
* when this PR removed the libical check and the `HAVE_RFC7986_COLOR`, the fallback implementation started being compiled in, because `HAVE_RFC7986_COLOR` was no longer defined, and shadowed the real library function
* we started using the fallback instead of the real library function, and started seeing the different behaviour
* which means the fix really is just cleaning up the old fallback function, like I noticed in my original review

I'll just push a commit that does that, since I'm already this deep anyway...

File: imap/ical_support.h (hunk 1 above)
Comment (elliefm) [reply imap/ical_support.h:213]: I wound up removing it myself!

**Comment by elliefm** [comment]
@rsto I guess you'd better review this now, since I've got commits on the branch now myself

**Comment by elliefm** [comment]
This fixes #4071

**Review by rsto — APPROVED** [review]

**Pull request merged**